lndmanage/lib/node.py

```diff
@@ -44,26 +47,33 @@
 OPEN_EXPIRY_TIME_MINUTES = 8
 
 
-class Node(object):
-    """Bare node object with attributes."""
-    def __init__(self):
-        self.alias = ''
-        self.pub_key = ''
-        self.total_capacity = 0
-        self.total_local_balance = 0
-        self.total_remote_balance = 0
-        self.total_channels = 0
-        self.num_active_channels = 0
-        self.num_peers = 0
-        self.total_satoshis_received = 0
-        self.total_satoshis_sent = 0
-        self.total_private_channels = 0
-        self.total_active_channels = 0
-        self.blockheight = 0
-
-
-class LndNode(Node):
-    """Implements the node interface for LND."""
+class LndNode:
+    """Implements a synchronous/asynchronous interface to an lnd node."""
```

Comment:
This is of course subjective but personally I think that this (refactor, making it async-capable) could be split into a separate individual PR which does not introduce further user-facing features.

This would make the changes a lot cleaner, easier to isolate bugs, follow up tests, etc. Especially since this is at the same time touching a large portion of files, restructuring all tests, and introducing major new functionality/features.

Also easier to follow for others ^^

Reply:
Thanks for the feedback! Right, I'll need to split up things and reorganize. At the moment I'm trying to do a MVP and glue the components together to see if things are working the way I see it, will try to keep it cleaner from the get go.

Reply:
I guess all good while you're still working out how you want it to be done!
What usually works well with things like this is to make a more frequent and smaller commits, where each should only be concerned with at most one concern, module, and/or "thing" - it's less work and time to squash them together than it is to break up larger ones later :)